lib/virtio/virtqueue.c

```diff
@@ -707,6 +819,12 @@ static int virtqueue_navail(struct virtqueue *vq)
 {
 	uint16_t avail_idx, navail;
 
+#ifdef VIRTIO_D_CACHE
+	metal_cache_invalidate(VRING_AVAIL(vq),
```

Comment:
here seems that you can just invalidate vq->vq_ring.avail->idx

Reply:
ACK